Bottom: 7a9147bbf4814b4ce56b28a03dae65dad5aa5118
Top:    0f06403259dd554d2276f9babe4108fe36448658
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-04-17 07:43:05 +0800

Refresh of tuntap-simplify-error-handling

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index e180bda..65e4afef 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1686,6 +1686,9 @@ static struct sk_buff *tun_build_skb(struct tun_struct *tun,
 			goto err_xdp;
 		}
 	}
+	rcu_read_unlock();
+	preempt_enable();
+
 
 	skb = build_skb(buf, buflen);
 	if (!skb) {
@@ -1696,9 +1699,6 @@ static struct sk_buff *tun_build_skb(struct tun_struct *tun,
 	skb_reserve(skb, pad - delta);
 	skb_put(skb, len + delta);
 
-	rcu_read_unlock();
-	preempt_enable();
-
 	return skb;
 
 err_xdp:
